cli/src/infragpt/llm_adapter.py

```
@@ -1,332 +1,223 @@
+            model_string: Provider:model format (e.g., "openai:gpt-4o")
+            api_key: API key for the provider
+            verbose: Enable verbose logging
+        """
+        self.model_string = model_string
+        self.api_key = api_key
+        self.verbose = verbose
+        
+        try:
+            self.provider = LLMRouter.create_provider(model_string, api_key)
+        except Exception as e:
+            raise ValidationError(f"Failed to initialize LLM provider: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)